### DISCUSSION TITLE
pnpm exec to run command sequentially

### DISCUSSION THREAD

**Discussion by privatenumber** [pnpm/pnpm#3712] [Q&A] [answered]
How can I make `pnpm exec` run the same command _sequentially_ in the sub-packages of a monorepo?

The [documentation of the `--parallel` flag](https://pnpm.io/cli/exec#--parallel) seems to imply that without the flag, the commands should run sequentially, however, I am finding that they run in parallel (kind of).

```sh
$ pnpm -r exec -- node -e 'setTimeout(() => console.log(process.env.PNPM_PACKAGE_NAME, new Date()), 5000)'

root-package 2021-08-25T23:52:04.413Z
subpackage-a 2021-08-25T23:52:04.414Z
subpackage-b 2021-08-25T23:52:04.420Z
subpackage-c 2021-08-25T23:52:04.420Z
subpackage-d 2021-08-25T23:52:09.498Z
```

Subpackages a, b, and c are logged at the same time (runs in parallel), and subpackage d is logged with a 5 second delay (sequential).

I would like for a, b, c, and d to log with a 5 second delay in between them all.

**Comment by zkochan** [comment, marked as answer]
You can run it with `--workspace-concurrency=1`

**Reply by privatenumber** [reply]
Perfect! Thank you for the prompt answer.

Curious, what does `--parallel` do if it's parallel by default?
